app/src/main/java/space/taran/arknavigator/mvp/model/repo/kind/DocumentKindFactory.kt

```diff
@@ -37,11 +37,14 @@ object DocumentKindFactory : ResourceKindFactory<ResourceKind.Document> {
         val totalPages = pdfRenderer.pageCount
         val pages = if (totalPages > 0) totalPages else null
 
-        return ResourceKind.Document(pages)
+        if (pages != null) {
+            return ResourceKind.Document(pages.toLong())
+        }
+        return ResourceKind.Document(null)
```

Comment:
If this resource can't be determined as `ResourceKind.Document` then just return `null`.